slave/manifests/site.pp

```diff
@@ -41,6 +41,25 @@
   notice("No ssh_keys defined. You should probably have at least one.")
 }
 
+# Setup SSH known hosts
+file { '/home/jenkins-slave/.ssh/' :
+  ensure => 'directory',
+  mode   => '644',
+  owner  => 'jenkins-slave',
+  group  => 'jenkins-slave',
+  require => User['jenkins-slave'],
+}
+
+$host_keys = hiera('ssh_host_keys', [])
+file { '/home/jenkins-slave/.ssh/known_hosts':
+  ensure => 'present',
+  mode => '0644',
+  owner => 'jenkins-slave',
+  group => 'jenkins-slave',
+  content => template('slave_files/known_hosts.erb'),
+  require => User['jenkins-slave'],
```

Comment:
This needs to require File['/home/jenkins-slave/.ssh/'] which in turn relies on the User['jenkins-slave'] so you can just substitute it.